api/apps/geoprocessing/src/export/pieces-exporters/project-cost-surfaces.piece-exporter.ts

```diff
@@ -0,0 +1,140 @@
+import { isLeft } from 'fp-ts/lib/Either';
+import { Readable } from 'stream';
+import { EntityManager } from 'typeorm';
+import {
+  ExportPieceProcessor,
+  PieceExportProvider,
+} from '../pieces/export-piece-processor';
+import { ProjectsPuEntity } from '@marxan-jobs/planning-unit-geometry';
+import { ProjectCostSurfacesContent } from '@marxan/cloning/infrastructure/clone-piece-data/project-cost-surfaces';
+
+type CreationStatus = ProjectCustomFeature['creation_status'];
+
+type ProjectCostSurfacesSelectResult = {
+  id: string;
+  name: string;
+  min: number;
+  max: number;
+  is_default: boolean;
+};
+
+type CostSurfaceDataSelectResult = {
+  cost_surface_id: string;
+  cost: number;
+  projects_pu_id: number;
+};
+
+@Injectable()
+@PieceExportProvider()
+export class ProjectCostSurfacesPieceExporter implements ExportPieceProcessor {
+  private readonly logger: Logger = new Logger(
+    ProjectCostSurfacesPieceExporter.name,
+  );
+
+  constructor(
+    private readonly fileRepository: CloningFilesRepository,
+    @InjectEntityManager(geoprocessingConnections.apiDB)
+    private readonly apiEntityManager: EntityManager,
+    @InjectEntityManager(geoprocessingConnections.default)
+    private readonly geoprocessingEntityManager: EntityManager,
+  ) {}
+
+  isSupported(piece: ClonePiece, kind: ResourceKind): boolean {
+    return (
+      piece === ClonePiece.ProjectCostSurfaces && kind === ResourceKind.Project
+    );
+  }
+
+  async run(input: ExportJobInput): Promise<ExportJobOutput> {
+    const costSurfaces: ProjectCostSurfacesSelectResult[] = await this.apiEntityManager
+      .createQueryBuilder()
+      .select(['cs.id', 'cs.name', 'cs.min', 'cs.max', 'cs.is_default'])
+      .from('cost_surfaces', 'cs')
+      .where('cs.project_id = :projectId', { projectId: input.resourceId })
+      .execute();
+
+    const costSurfacesIds = costSurfaces.map((costSurface) => costSurface.id);
+    let costSurfaceData: CostSurfaceDataSelectResult[] = [];
+    let projectPusMap: Record<string, number> = {};
+    if (costSurfacesIds.length > 0) {
+      costSurfaceData = await this.geoprocessingEntityManager
+        .createQueryBuilder()
+        .select(['cost_surface_id', 'cost', 'projects_pu_id'])
+        .from('cost_surface_pu_data', 'scpd')
+        .where('cost_surface_id IN (:...costSurfacesIds)', {
+          costSurfacesIds,
+        })
+        .execute();
+
+      projectPusMap = await this.getProjectPusMap(input.resourceId);
+    }
+
+    const fileContent: ProjectCostSurfacesContent = {
+      costSurfaces: costSurfaces.map(({ id, ...costSurface }) => ({
+        ...costSurface,
+        data: costSurfaceData
+          .filter(
+            (data: CostSurfaceDataSelectResult) => data.cost_surface_id === id,
+          )
+          .map(({ cost_surface_id, projects_pu_id, ...data }) => {
+            const puid = projectPusMap[projects_pu_id];
+            return { puid, ...data };
+          }),
+      })),
+    };
```

Comment:
just a note on style - my recommendation for these kinds of "augmentations" is to use named functions to make what we're doing via array operations (filter, map, etc.), spreads and so on clearer.

In practice here we're combining a lookup of cost surface costs with the setter part of a lens, using lower-level primitives: which is ok and works, but makes even a tiny snippet of code like this less legible than something like `const fileContent: ProjectCostSurfacesContent = addCostDataToCostSurfaces(costSurfaces, costSurfaceCostData)`

but no need to do anything here! (and as always, these kinds of style comments can be highly opinable)